src/Microsoft.VisualStudio.VsInteractiveWindow/IVsInteractiveWindowFactory2.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.InteractiveWindow.Shell
+{
+    public interface IVsInteractiveWindowFactory2: IVsInteractiveWindowFactory 
+    {
+        IVsInteractiveWindow Create(Guid providerId, int instanceId, string title, IInteractiveEvaluator evaluator, IVsInteractiveWindowDecorator decorator, __VSCREATETOOLWIN creationFlags = 0);
```

Comment:
I don't think having a ton of parameters is a problem, as long as we use optional parameters. 

---
In reply to: [91172076](https://github.com/dotnet/interactive-window/pull/23#discussion_r91172076) [](ancestors = 91172076,90067669)